drivers/flash/flash_stm32.c

```diff
@@ -257,7 +272,7 @@ static int stm32_flash_init(struct device *dev)
 
 	k_sem_init(&p->sem, 1, 1);
 
-	return flash_stm32_write_protection(dev, false);
+	return flash_stm32_write_protection(dev, true);
```

Comment:
What is the reason to change this here ? Is this related with F7 support ?

Reply:
I just think that set flash protection false in init api is not necessary.

Reply:
Please push in a separate commit and a separate PR, this is not the right place to change that